### PR TITLE
Network: add games argument to GetDataPackage

### PR DESCRIPTION
**Pull request by Berserker66**

### CODE REVIEW

File: MultiServer.py [hunk 1]
```diff
@@ -1474,6 +1475,14 @@ async def process_client_cmd(ctx: Context, client: Client, args: dict):
             package["games"] = games
             await ctx.send_msgs(client, [{"cmd": "DataPackage",
                                           "data": package}])
+        elif "games" in args:
```
Comment (black-sliver) [inline MultiServer.py:1478]: Think it would be better to
1. prefer the new version (`if "games" in args and args["games"]: new_behaviour; else: old_behaviour`)
2. merge the code in a way that `args["games"]` simply overwrites `games` and everything else stays the same

File: Utils.py [hunk 2]
```diff
@@ -55,8 +55,7 @@ def snes_to_pc(value):
 
 
 def cache_argsless(function):
-    if function.__code__.co_argcount:
-        raise Exception("Can only cache 0 argument functions with this cache.")
+    assert function.__code__.co_argcount, "Can only cache 0 argument functions with this cache."
```
Comment (black-sliver) [inline Utils.py]: should this really be in this PR? i don't really mind, but maybe unrelated cleanup commits should go in separate PRs  going forward if hey are not required by the change so it's easier to git blame (especially if the PR ends up being squashed)

**Review by black-sliver — COMMENTED** [review]
Another thing I noticed is that using room_info["games"] to create the include/exclude would exclude "Archipelago". I hard-coded "Archipelago" to ignore if it's being played or not in my client code, but that reads like an ugly hack.

**Comment by black-sliver** [comment]
Took me a while to understand the commit message (did not check the implementation after leaving my comment, assuming either discussion or code change to follow).

The solution of dropping summed version is rather odd. I'd just deprecate it in the docs but still send it and once exclude is gone, drop it from the docs and network_data_package in one go. Could then even simplify the code to `if name in set(...) or ("games" not in args: and name not in exclusions}` and drop the `or ...` once exlude is gone.

But I may be over-engineering the simplifications here :D

**Comment by Berserker66** [comment]
I figured I'd make it so that if you send the new format you also get back the new format so you can update your client in one go, without potentially forgetting about that and having to go change it again.

**Review by KonoTyran — APPROVED** [review]

**Comment by KonoTyran** [comment]
Github is still saying there are mege conflicts here.
